src/Publisher/FilesystemPublisher.php

```diff
@@ -174,8 +204,35 @@ protected function saveToPath($content, $filePath)
         // Move the temporary file to the desired location (prevents unlocked files from being read during write)
         $publishPath = $this->getDestPath() . DIRECTORY_SEPARATOR . $filePath;
         Filesystem::makeFolder(dirname($publishPath));
+        $successWithPublish = rename($temporaryPath, $publishPath);
+        if ($successWithPublish) {
+            if (FilesystemPublisher::config()->get('use_gzip_compression')) {
+                $publishPath = $this->compressFile($publishPath);
+            }
+        }
 
-        return rename($temporaryPath, $publishPath);
+        return file_exists($publishPath);
```

Comment:
I feel like if the gzip compression fails we shouldn't return false - I think it should be an optional step.